reana_server/scheduler.py

```diff
@@ -64,22 +65,28 @@ def check_predefined_conditions():
 
 def doesnt_exceed_max_reana_workflow_count():
     """Check upper limit on running REANA batch workflows."""
+    doesnt_exceed = True
     try:
-        running_workflows = Workflow.query.filter(
-            or_(
-                Workflow.status == RunStatus.pending,
-                Workflow.status == RunStatus.running,
+        running_workflows = (
+            Session.query(func.count())
```

Comment:
Just curious, how does it know which table it needs to query? Should we point that we want to query against `Workflow` somehow?

Reply:
It knows by the `filter()`, but perhaps we could do write `func.count(Workflow.id_)` to make it very explicit there as well... I can retest the speed of the two.

Reply:
Just for the record, I tested `func.count(Workflow.id_)`, the generated query look good:

```
SELECT count(*) AS count_1 
FROM __reana.workflow 
WHERE __reana.workflow.status = %(status_1)s OR __reana.workflow.status = %(status_2)s

vs

SELECT count(__reana.workflow.id_) AS count_1 
FROM __reana.workflow 
WHERE __reana.workflow.status = %(status_1)s OR __reana.workflow.status = %(status_2)s
```

The performance (on a small table) was a bit slower though. The differences were not much, of the order of microseconds, but for extra safety for big tables, perhaps we could leave the current version...  I guess `COUNT(*)` might be slightly more optimised than `COUNT(somecolumn)` in RDBMS implementations in general.